### PR TITLE
Cocoon

### PR DESCRIPTION
**Pull request by VatsalMewada**

### CODE REVIEW

File: app/assets/config/admintask.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+function myFunction() {
```
Comment (swati-bacancy) [inline app/assets/config/admintask.js]: is any need of this function?
new line at the end of the file

File: app/assets/config/manifest.js [hunk 2]
```diff
@@ -1,2 +1,5 @@
 //= link_tree ../images
 //= link_directory ../stylesheets .css
+function myFunction() {
```
Comment (swati-bacancy) [inline app/assets/config/manifest.js]: is any need of this function?
new line at the end of the file

File: app/assets/stylesheets/admin.scss [hunk 3]
```diff
@@ -1,3 +1,31 @@
 // Place all the styles related to the Admin controller here.
 // They will automatically be included in application.css.
 // You can use Sass (SCSS) here: https://sass-lang.com/
+th, td {
```
Comment (swati-bacancy) [inline app/assets/stylesheets/admin.scss]: 2 space indentation

File: app/assets/stylesheets/modifysubject.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the modifysubject controller here.
```
Comment (swati-bacancy) [inline app/assets/stylesheets/modifysubject.scss]: remove unneccessary files if you are not adding any css same for all the js files

File: app/controllers/admin_controller.rb [hunk 5]
```diff
@@ -1,37 +1,67 @@
+# frozen_string_literal: true
+
+# admincontroller
 class AdminController < ApplicationController
-  def index
-  end
+  def index; end
```
Comment (swati-bacancy) [inline app/controllers/admin_controller.rb]: remove this function if no code in the function

File: app/controllers/admin_controller.rb [hunk 6]
```diff
@@ -1,37 +1,67 @@
...
-  end
-  
   def new
     @user = User.new
+    @users = User.all
   end
 
   def create
     @user = User.new(info_params)
+    @user.branch_id = params[:branch_id]
```
Comment (swati-bacancy) [inline app/controllers/admin_controller.rb]: please use nested form for this

File: app/controllers/admin_controller.rb [hunk 7]
```diff
@@ -1,37 +1,67 @@
...
     if @user.save
-      redirect_to admin_index_path
+      if @user.status.to_i == 2
```
Comment (swati-bacancy) [inline app/controllers/admin_controller.rb]: why this condition?

File: app/controllers/admin_controller.rb [hunk 8]
```diff
@@ -1,37 +1,67 @@
...
+        @user.add_role :faculty
+        redirect_to new_admin_path(id: 'faculty')
+      end
+    else
+      render :new, id: 'faculty'
     end
   end
 
-  def edit
+  def filltable
```
Comment (swati-bacancy) [inline app/controllers/admin_controller.rb]: what is the use of this function?

File: app/controllers/exams_controller.rb [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  def index
+    @exams = current_user.exams
+  end
+
+  def new
+    @exam = Exam.new
+  end
+
+  def create
+    @exam = Exam.new(info_params)
```
Comment (swati-bacancy) [inline app/controllers/exams_controller.rb]: you can use current_user.exams.build and also nested form

File: app/controllers/exams_controller.rb [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    @exam.user_id = current_user.id
+    @exam.subject_id = params[:subject_id]
+    if @exam.save!
+      flash[:notice] = 'Exam added Successfully'
```
Comment (swati-bacancy) [inline app/controllers/exams_controller.rb]: redirect and flash in same line

File: app/controllers/exams_controller.rb [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    #  byebug
+    @exam = Exam.find(params[:exam][:id])
+    if @exam.update(info_params)
+      flash[:notice] = 'Exam updated Successfully'
+      redirect_to exams_path
+    end
+  end
+
+  def destroy
+    @exam = Exam.find(params[:id])
```
Comment (swati-bacancy) [inline app/controllers/exams_controller.rb]: please add one function for findexam

File: app/controllers/faculties_controller.rb [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+      flash[:notice] = 'Faculty updated Successfully'
+      redirect_to new_faculty_path
+    else
+      flash[:alert] = 'Something went wrong!'
+    end
+  end
+
+  def create
+    @user = User.new(info_params)
+    a = params[:branch_id]
```
Comment (swati-bacancy) [inline app/controllers/faculties_controller.rb]: what is a use meaning full name

File: app/controllers/options_controller.rb [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+  def show; end
+
+  def edit
+    # byebug
+    @option = Option.find(params[:id])
+  end
+
+  def update
+    # byebug
+    @option = Option.find(params[:id])
```
Comment (swati-bacancy) [inline app/controllers/options_controller.rb]: create new function for find question.
change in all the controllers

File: app/controllers/options_controller.rb [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+      flash[:alert] = 'Something went wrong!'
+    end
+  end
+
+  def info_params
+    params.required(:option).permit(:opt, :question_id)
+  end
+
+  def destroy
+    byebug
```
Comment (swati-bacancy) [inline app/controllers/options_controller.rb]: remove byebug
do not push byebug, debugger and binding.pry in git

File: app/helpers/admin_helper.rb [hunk 15]
```diff
@@ -1,2 +1,5 @@
+# frozen_string_literal: true
```
Comment (swati-bacancy) [inline app/helpers/admin_helper.rb]: remove helpers file is no any use

File: app/models/branch.rb [hunk 16]
```diff
@@ -0,0 +1,4 @@
+class Branch < ApplicationRecord
+  has_many :users
```
Comment (swati-bacancy) [inline app/models/branch.rb]: add dependent destroy

File: app/views/admin/index.html.erb [hunk 17]
```diff
@@ -1,20 +1,25 @@
...
-<%if current_user%>
-<!--<%= @user = current_user %>
-<%= @user.email%>-->
 
-<%= link_to 'Edit Profile', edit_user_registration_path%>
-<%= link_to 'sign out', destroy_user_session_path, method: :delete %>
-<br>
-<%= link_to 'Add Faculty', admin_new_path%>
-<%= link_to 'Add Students', admin_new_path%>
+    <%if current_user%>
```
Comment (swati-bacancy) [inline app/views/admin/index.html.erb]:
```suggestion
    <% if current_user %>
```

File: app/views/admin/index.html.erb [hunk 18]
```diff
@@ -1,20 +1,25 @@
...
+    <!--<%= @user = current_user %>
```
Comment (swati-bacancy) [inline app/views/admin/index.html.erb]: you can directly use current user no need for @user

File: app/views/admin/index.html.erb [hunk 19]
```diff
@@ -1,20 +1,25 @@
...
-<%end%>
+    <%else%>
+    <script type="text/javascript">
+        window.location.href = "/" // put your correct path in a string here
+    </script>
+    <!--<%= link_to 'sign up', new_user_registration_path %>-->
+    <%= link_to 'sign in', new_user_session_path %>
+    <%end%>
 
+</center>
```
Comment (swati-bacancy) [inline app/views/admin/index.html.erb]: eol

File: app/views/admin/new.html.erb [hunk 20]
```diff
@@ -1,42 +1,230 @@
...
-    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
-  </div>
-
-  <div class="actions">
-    <%= f.submit "Sign up" %>
-  </div>
-<% end %>
+<%= link_to 'Back', admin_index_path%>
+<script>
+  function myFunction(category) {
```
Comment (swati-bacancy) [inline app/views/admin/new.html.erb]: what is the need of this function?

File: app/views/admin/new.html.erb [hunk 21]
```diff
@@ -1,42 +1,230 @@
...
+    alert(category);
+  }
+</script>
+
+<% if params[:id]=="student" %>
```
Comment (swati-bacancy) [inline app/views/admin/new.html.erb]: single quotes

File: app/views/admin/new.html.erb [hunk 22]
```diff
@@ -1,42 +1,230 @@
...
+
+<select id="category" onChange="alert(this.value);">
```
Comment (swati-bacancy) [inline app/views/admin/new.html.erb]: please reflector this file

File: app/views/admin/new.html.erb [hunk 23]
```diff
@@ -1,42 +1,230 @@
...
+
+
+    </table>
+  </center>
+</div>
+
+
+<% else%>
+<h2>Else Part</h2>
+<%end%>
```
Comment (swati-bacancy) [inline app/views/admin/new.html.erb]: eol

File: app/views/devise/registrations/edit.html.erb [hunk 24]
```diff
@@ -1,144 +1,150 @@
...
-  </div>
+<div class="field">
+  <%= f.label :email %><br />
+  <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+</div>
 
-  <% if devise_mapping.confirmable? && resource.pending_reconfirmation? %>
-    <div>Currently waiting confirmation for: <%= resource.unconfirmed_email %></div>
+<% if devise_mapping.confirmable? && resource.pending_reconfirmation? %>
+<div>Currently waiting confirmation for: <%= resource.unconfirmed_email %></div>
```
Comment (swati-bacancy) [inline app/views/devise/registrations/edit.html.erb]: 2 space indentation on all the file

File: test/controllers/addquestion_controller_test.rb [hunk 25]
```diff
@@ -0,0 +1,34 @@
+require 'test_helper'
+
+class AddquestionControllerTest < ActionDispatch::IntegrationTest
+  test "should get index" do
+    get addquestion_index_url
+    assert_response :success
```
Comment (swati-bacancy) [inline test/controllers/addquestion_controller_test.rb]: if you are not writting the test case the please remove this files

**Comment by VatsalMewada** [comment]
not needed

**Pull request closed without merging**